### PR TITLE
chore: sealing pipeline: Remove adapter code (storage refactors part 2)

### PR DESCRIPTION
**Pull request by magik6k**
## Related Issues
Part 2 of https://github.com/filecoin-project/lotus/issues/8793

## Proposed Changes
Remove redundant adapter boilerplate from the `storage/` package. Use the API directly.

Also adds cbor marshalers to TipSetKey which match the now removed TipSetTokens. Confirmed that it's still compatible with existing metadata on my SP setup.

### CODE REVIEW

**Review by geoff-vball — COMMENTED** [review]
Just one comment to be addressed. Everything else looks good :)

File: markets/storageadapter/ondealsectorcommitted.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type eventsCalledAPI interface {
 }
 
 type dealInfoAPI interface {
-	GetCurrentDealInfo(ctx context.Context, tok pipeline.TipSetToken, proposal *market.DealProposal, publishCid cid.Cid) (pipeline.CurrentDealInfo, error)
+	GetCurrentDealInfo(ctx context.Context, tok types.TipSetKey, proposal *market.DealProposal, publishCid cid.Cid) (pipeline.CurrentDealInfo, error)
```
Comment (geoff-vball) [inline markets/storageadapter/ondealsectorcommitted.go:31]: tsk

File: markets/storageadapter/ondealsectorcommitted_test.go [hunk 2]
```diff
@@ -574,7 +574,7 @@ type mockDealInfoAPI struct {
 	Err2             error
 }
 
-func (m *mockDealInfoAPI) GetCurrentDealInfo(ctx context.Context, tok pipeline.TipSetToken, proposal *market.DealProposal, publishCid cid.Cid) (pipeline.CurrentDealInfo, error) {
+func (m *mockDealInfoAPI) GetCurrentDealInfo(ctx context.Context, tok types.TipSetKey, proposal *market.DealProposal, publishCid cid.Cid) (pipeline.CurrentDealInfo, error) {
```
Comment (geoff-vball) [inline markets/storageadapter/ondealsectorcommitted_test.go:577]: tsk

File: storage/miner.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ type fullNodeFilteredAPI interface {
 	// Call a read only method on actors (no interaction with the chain required)
 	StateCall(context.Context, *types.Message, types.TipSetKey) (*api.InvocResult, error)
 	StateMinerSectors(context.Context, address.Address, *bitfield.BitField, types.TipSetKey) ([]*miner.SectorOnChainInfo, error)
-	StateSectorPreCommitInfo(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (miner.SectorPreCommitOnChainInfo, error)
+	StateSectorPreCommitInfo(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (*miner.SectorPreCommitOnChainInfo, error)
 	StateSectorGetInfo(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (*miner.SectorOnChainInfo, error)
 	StateSectorPartition(ctx context.Context, maddr address.Address, sectorNumber abi.SectorNumber, tok types.TipSetKey) (*lminer.SectorLocation, error)
```
Comment (geoff-vball) [inline storage/miner.go:87]: tsk

Comment (magik6k) [reply storage/miner.go:86]: no tsk?

Comment (geoff-vball) [reply storage/miner.go:86]: You can change it or not before merge. Everything else looks good now :)

**Pull request merged**